subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/delegates/ProjectDelegate.kt

```diff
@@ -244,10 +246,14 @@ abstract class ProjectDelegate : Project {
     override fun components(configuration: Action<in SoftwareComponentContainer>) =
         delegate.components(configuration)
 
+    @Deprecated("Use layout.buildDirectory instead", ReplaceWith("layout.buildDirectory.set(path)"))
     override fun setBuildDir(path: File) {
+        @Suppress("DEPRECATION")
         delegate.buildDir = path
     }
 
+    @Suppress("DEPRECATION")
+    @Deprecated("Use layout.buildDirectory instead", ReplaceWith("layout.buildDirectory.set(file(path))"))
     override fun setBuildDir(path: Any) =
         delegate.setBuildDir(path)
```

Comment:
❔ Can't the deprecation suppression be reduced to the call to the delegate as in `setBuildDir` above?


```suggestion
    @Deprecated("Use layout.buildDirectory instead", ReplaceWith("layout.buildDirectory.get()"))
    override fun setBuildDir(path: Any) =
        @Suppress("DEPRECATION")
        delegate.setBuildDir(path)
```